htsget-search/src/storage/aws.rs

```diff
@@ -53,12 +53,15 @@ impl AwsS3Storage {
     AwsS3Storage { client, bucket }
   }
 
-  pub async fn new_with_default_config(bucket: String, endpoint: Option<String>) -> Self {
+  pub async fn new_with_default_config(bucket: String, endpoint: Option<String>, path_style: bool) -> Self {
     let sdk_config = aws_config::load_from_env().await;
-    let mut s3_config_builder = aws_sdk_s3::config::Builder::from(&sdk_config);
-    s3_config_builder.set_endpoint_url(endpoint); // For local S3 storage, i.e: Minio
 
-    let client = s3_config_builder.build();
+    let client = aws_sdk_s3::config::Builder::from(&sdk_config)
+       .set_endpoint_url(endpoint)
+       .clone()
```

Comment:
You could avoid a `.clone()` here by reverting this to its previous form (where `s3_config_builder` was assigned to a mutable variable) and using `set_force_path_style` instead. For example:
```rust
s3_config_builder.set_force_path_style(Some(path_style));
```